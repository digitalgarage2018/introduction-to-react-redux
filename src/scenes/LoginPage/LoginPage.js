/**
 * Created by albertogiovanelli on 15/05/18.
 */
import React, {Component} from 'react';
import Button from '../../components/Button/Button';
import {fakeLogin, getData} from '../../fakeLogin/fakeLogin';
import {login} from '../../actions/auth/authAction';
import {connect} from 'react-redux';


export class LoginPage extends React.Component {
    constructor() {
        super();
        this.onHandleChange = this.onHandleChange.bind(this);
    }

    state = {
        username: '',
        password: ''
    };

    async onSubmit(e) {
        e.preventDefault();
        try {
            const jwt = await fakeLogin(this.state);
            this.props.dispatch(login({
                accessToken: jwt
            }));
            this.props.history.push('/dashboard');
            const data = await this.getData(jwt);
            console.log("data", data);
        } catch (err) {
            console.log(`Something went wrong ${err}`);
        }
    };

    async getData(jwt) {
        return getData(jwt);
    }

    onHandleChange(e) {
        const {name, value} = e.target;
        this.setState({
            [name]: value
        }, () => console.log(this.state));
    }

    render() {
        return (
            <div className="login">
                <form className="login__container" onSubmit={(e) => this.onSubmit(e)}>
                    <input type="text" className="login__container--username" name="username" placeholder="Username..."
                           onChange={this.onHandleChange}/>
                    <input type="password" className="login__container--password" name="password"
                           placeholder="Password..." onChange={this.onHandleChange}/>
                    <Button type="submit" label="Login" className="login__container--button"
                    >Login</Button>
                </form>
            </div>
        );
    }
}


export default connect()(LoginPage);