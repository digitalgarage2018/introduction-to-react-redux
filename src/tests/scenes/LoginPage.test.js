import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import {LoginPage} from '../../scenes/LoginPage/LoginPage';

test('should render LoginPage correctly', () => {
    const wrapper = shallow(<LoginPage/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
});