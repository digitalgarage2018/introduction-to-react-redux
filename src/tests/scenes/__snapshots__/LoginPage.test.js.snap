// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div
  className="login"
>
  <form
    className="login__container"
    onSubmit={[Function]}
  >
    <input
      className="login__container--username"
      name="username"
      onChange={[Function]}
      placeholder="Username..."
      type="text"
    />
    <input
      className="login__container--password"
      name="password"
      onChange={[Function]}
      placeholder="Password..."
      type="password"
    />
    <Button
      className="login__container--button"
      disabled={false}
      label="Login"
      type="submit"
    >
      Login
    </Button>
  </form>
</div>
`;
